File: phoenix/keycloakuser/src/main/java/com/phoenix/dto/Userdto.java
package com.phoenix.dto;

import com.phoenix.model.UserMysql;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import lombok.*;

import java.util.Date;
import java.util.List;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Userdto {
    private String firstName;
    private String lastName;
    private String email;
    private String username;
    private String password;
    private List<String> realmRoles;
    private byte[] image;

    private Long phone;
    private String jobTitle;
    private Date dateDebutContrat;
    private Date dateFinContrat;
    private boolean usertypemanager;

    private UserMysql manager;


}

File: phoenix/phoenixfront/src/app/models/agents/Userdto.ts
export interface Userdto {
    firstName?: string;
    lastName?: string;
    email?: string;
    username?: string;
    password?: string;
    realmRoles?: string[] | null;
    image?: string | null;
    phone?: number;
    jobTitle?: string;
    dateDebutContrat?: Date;
    dateFinContrat?: Date;
    usertypemanager?: boolean;
    manager?: Userdto;
    associatedProds?: number;
    returnedProds?: number;
    soldProds?: number;
  
}